### PR TITLE
removed(plugins): plugins no longer work with manifest files

### PR DESCRIPTION
**Pull request by jdalsem**
- [ ] add new ElggPlugin tests
- [x] add generic plugin composer test? similar to ManifestParsingTest
- [x] elgg-plugin.php docs about plugin section
- [ ] update plugin dependencies docs
- [ ] fix plugin dependencies report
- [ ] assert plugin priority dependency

### CODE REVIEW

File: docs/appendix/upgrade-notes/3.x-to-4.0.rst [hunk 1]
```diff
@@ -49,6 +49,31 @@ Miscellaneous changes
...
+
+php version requirement -> composer require
+php extension requirement -> composer require
+plugin conflicts -> composer conflicts
+plugin requirement -> elgg-plugin
+plugin position requirement -> elgg-plugin
+
+plugin version -> elgg-plugin
+plugin activate on install -> elgg-plugin
+plugin name -> activate on install
```
Comment (iionly) [inline docs/appendix/upgrade-notes/3.x-to-4.0.rst:67]: plugin name -> ???

Comment (jdalsem) [reply docs/appendix/upgrade-notes/3.x-to-4.0.rst:67]: uh... thanks for reviewing :)

Comment (iionly) [reply docs/appendix/upgrade-notes/3.x-to-4.0.rst]: I've missed it before but the plugin name line was already there before. So, it's listed twice now.

As far as I know the "suggests" tag was already deprecated in Elgg 3.3. May I suggest to include the usage of "suggest" in composer.json in any docs (if any documentation about usage of composer.json exists / is planned to be added).

**Comment by jdalsem** [comment]
Replaced by #13327

**Pull request closed without merging**